src/compiler/checker.ts

```diff
@@ -210,6 +210,10 @@ namespace ts {
             getSuggestionForNonexistentProperty,
             getSuggestionForNonexistentSymbol,
             getBaseConstraintOfType,
+            getJsxNamespaceSymbol: node => {

```

Comment:
Also, why not just return the JSX namespace itself and look up the original entity in the `exports` for that symbol?